tidy-clustering/README.Rmd

```diff
@@ -0,0 +1,296 @@
+
+    +  A cluster structure being stable doesn't necessarily mean it is meaningful.
+    See e.g. [this tweet](https://twitter.com/AndrewZalesky/status/1363741266761027585)
+    
+    
+3.  **Enrichment in known labels:** (See *semi-supervised learning* above)
+
+4.  **Statistical significance/generative models:** If our clustering method 
+places model assumptions on how the data was generated, we can formulate a notion
+of statistically signficant clusters.  
+
+    +  A good example of this is "model-based clustering", which typically assumes
+    the data is generated from a mixture of multivariate Gaussians.  We can then
+    estimate the probability that a particular object came from a particular 
+    distribution in the mixture.
+    
+    
+## Other details
+
+#### Partitions versus Extractions
+
+In many clustering algorithms, each object is placed into **exactly one** cluster.
+This is called a **partition**.
+
+However, some algorithms allow for overlapping clusters, i.e. objects belonging to 
+multiple clusters - or for some objects to be "background" and have no cluster 
+membership at all.
+
+(This is sometimes called **community detection** or **cluster extraction**)
+
+#### What are the variables, what are the samples?
+
+Consider the example of data consisting of many gene expression measurements
+for several individual subjects.
+
+From a statistical perspective, we'd typically regard the subjects as samples
+from a population, and the gene expression levels as variables being studied.
+
+When we cluster this data, sometimes we look for clusters of **subjects**, i.e.,
+people whose gene signature is similar.  Sometimes we look for clusters of **genes**,
+i.e., genes that activate or deactivate together.
+
+Non-statistical algorithms (e.g. kmeans) are agnostic to what set of objects we 
+cluster on.  In statistical algorithms, your model assumptions need to match
+your notion of samples/variables.
+
+#### Beyond the similarity matrix
+
+In the vast majority of cases, all you need for a clustering algorithm is a 
+*similarity matrix*; that is, entry $(i,j)$ of the matrix is the pairwise similarity 
+measure between object $i$ and object $j$.
+
+Of course, the choice of similarity measure matters enormously.  But the point is
+you only need pairwise similarities to run e.g. kmeans.
+
+Some approaches, though, need more information.  Statistical approaches in particular
+often estimate nuisance variables (like variance) from multiple measurements before
+collapsing the data into pairwise similarities.
+
+
+# Inclusion in tidymodels
+
+Here's an overview of how I see unsupervised learning fitting into a tidymodels
+framework.
+
+## Recipes and Workflow
+
+These steps can remain the same; there's no reason unsupervised methods need
+different preprocessing options/structure.
+
+## `parsnip`-like models
+
+Obviously, any and all clustering methods would have to be implemented as model
```

Comment:
I might take out "obviously" here because if a goal is for folks to be able to use these as preprocessing steps, then building this out using a recipes framework might be worth considering (rather than parsnip).

Reply:
I edited the sentence, but I disagree with using the recipes framework.

Even if an unsupervised procedure _informs_ a preprocessing, the application of the clustering method is not itself a preprocessing step.  The data is not altered by the search procedure.

Besides which, I think things could get a little confusing when clustering is the end point in and of itself - would it appear as a recipe with no "baking" step?  That feels strange to me.